profiling/src/collections/string_table/mod.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::collections::identifiable::{Id, StringId};
+use crate::iter::{IntoLendingIterator, LendingIterator};
+use datadog_alloc::{Allocator, ChainAllocator, VirtualAllocator};
+use std::alloc::Layout;
+
+type Hasher = core::hash::BuildHasherDefault<rustc_hash::FxHasher>;
+type HashSet<K> = indexmap::IndexSet<K, Hasher>;
+
+/// Holds unique strings and provides [StringId]s that correspond to the order
+/// that the strings were inserted.
+pub struct StringTable {
+    /// The bytes of each string stored in `strings` are allocated here.
+    bytes: ChainAllocator<VirtualAllocator>,
+
+    /// The ordered hash set of unique strings. The order becomes the StringId.
+    /// The static lifetime is a lie, it is tied to the `bytes`, which is only
+    /// moved if the string table is moved e.g.
+    /// [StringTable::into_lending_iterator].
+    /// References to the underlying strings should generally not be handed,
+    /// but if they are, they should be bound to the string table's lifetime
+    /// or the lending iterator's lifetime.
+    strings: HashSet<&'static str>,
+}
+
+impl Default for StringTable {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl StringTable {
+    /// Creates a new string table, which initially holds the empty string and
+    /// no others.
+    pub fn new() -> Self {
+        // Christophe and Grégory think this is a fine size for 32-bit .NET.
```

Comment:
Minor: This is the kind of comment that tends to age poorly, e.g. maybe in some time we won't remember what this meant; may be worth expanding a bit ;)

Reply:
Thoughts on the reworded comment?